pkg/apiserver/serializer_test.go

```diff
@@ -0,0 +1,77 @@
+package apiserver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type stubSerializer struct {
+	supportedTypes []runtime.SerializerInfo
+}
+
+func newStubSerializer(supportedTypes []runtime.SerializerInfo) *stubSerializer {
+	return &stubSerializer{supportedTypes: supportedTypes}
+}
+
+func (s *stubSerializer) SupportedMediaTypes() []runtime.SerializerInfo {
+	return s.supportedTypes
+}
+
+func (s *stubSerializer) EncoderForVersion(serializer runtime.Encoder, gv runtime.GroupVersioner) runtime.Encoder {
+	return nil
+}
+
+func (s *stubSerializer) DecoderToVersion(serializer runtime.Decoder, gv runtime.GroupVersioner) runtime.Decoder {
+	return nil
+}
+
+func TestNoProtobufSerializerSupportedMediaTypes(t *testing.T) {
+	tt := []struct {
+		name                 string
+		originalContentTypes []runtime.SerializerInfo
+		wantContentTypes     []runtime.SerializerInfo
+		wantStrictlyEqual    bool
+	}{
+		{
+			name: "Wrapping an original Protobuf-only serializer should return no supported types",
+			originalContentTypes: []runtime.SerializerInfo{
+				{MediaType: runtime.ContentTypeProtobuf},
+			},
+			wantContentTypes: []runtime.SerializerInfo{},
+		},
+		{
+			name: "Wrapping an original serializer should return original types",
+			originalContentTypes: []runtime.SerializerInfo{
+				{MediaType: runtime.ContentTypeProtobuf},
+				{MediaType: runtime.ContentTypeJSON},
+				{MediaType: runtime.ContentTypeYAML},
+			},
+			wantContentTypes: []runtime.SerializerInfo{
+				{MediaType: runtime.ContentTypeJSON},
+				{MediaType: runtime.ContentTypeYAML},
+			},
```

Comment:
do we want to add the case json+yaml -> json+yaml ?

Reply:
Good catch! We do, please see the update.